### PR TITLE
Backport of #1442 to 2.20.1

### PR DESCRIPTION
**Pull request by jasperalbers**
cherry-picked commits from pull request #1442, resolved a minor merge conflict

### CODE REVIEW

**Comment by heplesser** [comment]
@jasperalbers Thank you for preparing v2.20.1. 

In addition to #1442, please also integrate #1416, #1421, #1422, #1427, #1428, #1430, which all corrected weaknesses in the code which we discovered while solving #1394.

Furthermore, #1449 needs to be solved and the resolving PR integrated into 2.20.1 before a release. Therefore, I have marked this PR as "blocked".

**Comment by heplesser** [comment]
@jasperalbers All necessary PRs have now been merged into master, so you can proceed with cherry-picking into this branch.

**Comment by jasperalbers** [comment]
@heplesser I was advised by @terhorstd to create a separate pull request onto nest-2.20.1 for each of the PRs that you mentioned. Here is a complete list:

for #1416 : new PR #1462
for #1421 : new PR #1463
for #1428 : new PR #1464
for #1458 : new PR #1465

Note: @terhorstd suggested to exclude #1422, #1427 and #1430 as they are not bugfixes but rather developments not necessary for a NEST version that won't be maintained in the future.

Additional note: PR #1458 addresses changes in two files that are not present in NEST 2.20, namely models/music_rate_in_proxy.h and models/music_rate_in_proxy.cpp. I assume that they are supposed to be added, which I did in #1465.

**Comment by heplesser** [comment]
@jasperalbers Ok with the independent PRs, I just hope we don't get problems due to dependencies between the different fixes. I have merge two of the new PRs and will merge the third once tests have gone through (timeout on Travis).

@terhorstd @jasperalbers I would feel much more comfortable also to integrate #1422, #1427 and #1430, since they all were part of one development stream that led to the fix of #1394.

**Comment by heplesser** [comment]
@terhorstd Did we arrive at a decision in the VC yesterday about whether to include #1422, #1427 and #1430 in 2.20.1? I simply don't remember, the unstable connections/system were distracting.

@suku248 @jakobj @jarsi  Do you have any opinion on this?

**Comment by heplesser** [comment]
@jasperalbers Working on #1272, we discovered a bug for which we should have a PR soon. That PR should be included in 2.20.1 as well.

**Comment by heplesser** [comment]
> @jasperalbers Working on #1272, we discovered a bug for which we should have a PR soon. That PR should be included in 2.20.1 as well.

There is now a PR solving #1272: #1481. It should be integrated into 2.20.1 once it has been merged into master.

**Comment by heplesser** [comment]
Decision during hackathon today: do **not** include #1422, #1427 and #1430.

**Comment by heplesser** [comment]
The following PRs need to be integrated here: #1472, #1481, #1501, #1514.

**Comment by jasperalbers** [comment]
#1502 can be ticked off the list (which isn't strictly necessary anymore, all corresponding PRs can be found under the milestone 2.20.1), and this PR itself needs reviews and merging.
I renamed the PR to make it consistent with all the other backporting PRs.

**Comment by heplesser** [comment]
@jasperalbers Please also integrate #1566, it is just a single line.

**Comment by jasperalbers** [comment]
@heplesser done in #1572.

**Comment by heplesser** [comment]
> @heplesser done in #1572.

@jasperalbers Thanks! Then we just need to wrap up the review of #1472 (@terhorstd),  resolve the merge conflicts and be ready for the final review of this PR?

**Comment by heplesser** [comment]
@jasperalbers I think we have all individual pieces for 2.20.1 in place now. Could you check and solve the merge conflicts? Then we can review and hopefully release soon.

**Comment by jasperalbers** [comment]
@heplesser great! I also don't see any loose ends. I rebased my heisenbug-fix-branch to the current nest-2.20.1 and now all conflicts should be solved.

File: doc/download.rst [hunk 1]
```diff
@@ -40,15 +40,13 @@ Live media is available in the OVA format, and is suitable, for example, for imp
...
-
-  `Checksum 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova.sha512sum>`_
+* `NEST Live Media 2.20.1 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.1.ova>`_
 
+  `Checksum 2.20.1 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.1.ova.sha512sum>`_
 
-* `NEST live media 2.16.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.16.0.ova>`_
-
-  `Checksum 2.16.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.16.0.ova.sha512sum>`_
+* `NEST Live Media 2.20.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.0.ova>`_
```
Comment (heplesser) [inline doc/download.rst:47]: The links to 2.20.0 should also move down to "Older versions".

Comment (terhorstd) [reply doc/download.rst]: I moved 2.20 to older versions.

**Comment by jasperalbers** [comment]
@suku248 nothing particular comes to my mind. Notice that only your 6 commits are the core of this PR, the rest is here just from rebasing to the current 2.20.1 branch and are already reviewed!

**Pull request merged**